scripts/src/java/org/oppia/android/scripts/maven/GenerateLicenseTexts.kt

```diff
@@ -0,0 +1,423 @@
+package org.oppia.android.scripts.maven
+
+import org.oppia.android.scripts.proto.CopyrightLicense
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.oppia.android.scripts.proto.PrimaryLinkType
+import org.w3c.dom.Document
+import org.w3c.dom.Element
+import java.io.File
+import java.io.FileInputStream
+import java.net.URL
+import javax.xml.parsers.DocumentBuilder
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.transform.OutputKeys
+import javax.xml.transform.Transformer
+import javax.xml.transform.TransformerFactory
+import javax.xml.transform.dom.DOMSource
+import javax.xml.transform.stream.StreamResult
+
+private const val MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE =
+  "maven_dependencies.textproto is not up-to-date"
+
+// List of chars to be escaped to parse XML properly.
+// Reference Link: https://www.liquid-technologies.com/XML/EscapingData.aspx
+private val escapeCharactersMap =
+  hashMapOf<Char, String>(
+    '<' to "&lt;",
+    '>' to "&gt;",
+    '\"' to "&quot;",
+    '\'' to "&apos;",
+    '&' to "&amp;",
+  )
+
+/**
+ * Script to extract the licenses for the third-party Maven dependencies (direct and indirect both)
+ * on which Oppia Android depends.
+ *
+ * Usage:
+ *   bazel run //scripts:generate_license_texts -- <path_to_directory_values>
+ *   <name_of_value_resource_file_1> <name_of_value_resource_file_2>
+ *   <name_of_value_resource_file_2> <name_of_value_resource_file_4>
+ *   <name_of_value_resource_file_2> <path_to_pb_file>
```

Comment:
I think all of these can be directly defined in the script.

Reply:
Done.